### ISSUE TITLE
InitialState should match fields type if value is an array

### ISSUE THREAD

**Issue by xcatliu** [erikras/redux-form#382]
It seems if an `initialState` value is an array, `redux-form` will handle it as a deep form:

```js
// SimpleForm.js
class SimpleForm extends Component {
}
export default reduxForm({
  form: 'simple',
  fields: [ 'items' ],
  initialValues: {
    'items': [ 'hello', 'world' ]
  }
})(SimpleForm);
```

`state.form.simple.items` expected to be

![image](https://cloud.githubusercontent.com/assets/5453359/11587176/97895c8a-9ab2-11e5-8dd8-27c8ec465355.png)

but actually is

![image](https://cloud.githubusercontent.com/assets/5453359/11587199/b5218bbe-9ab2-11e5-8ed4-04739c5b8797.png)

BTW, `onChange` can accept an array value correctly:

```js
// SimpleForm.js
class SimpleForm extends Component {
  componentDidMount() {
    this.props.fields.items.onChange([ 'new', 'value' ]);
  }
}
export default reduxForm({
  form: 'simple',
  fields: [ 'items' ],
})(SimpleForm);
```

![image](https://cloud.githubusercontent.com/assets/5453359/11587297/49245e5e-9ab3-11e5-9b47-7fc67bf57885.png)

Is it better to check `fields` when initialing values, so it can handling array value?

**Comment by erikras** [comment]
Yes, this is the same as #360. I'm working on a solution.

**Comment by erikras** [comment]
This should be fixed in [`v4.0.0`](https://github.com/erikras/redux-form/releases/tag/v4.0.0). Come and get it!
